sdk/src/client/secret/mod.rs

```diff
@@ -553,8 +553,10 @@ where
                 // We can only sign ed25519 addresses and block_indexes needs to contain the account or nft
                 // address already at this point, because the reference index needs to be lower
                 // than the current block index
-                if !input_address.is_ed25519() {
-                    Err(InputSelectionError::MissingInputWithEd25519Address)?;
+                match &input_address {
+                    Address::Ed25519(_) | Address::ImplicitAccountCreation(_) => {}
+                    Address::Restricted(restricted) if restricted.address().is_ed25519() => {}
+                    _ => Err(InputSelectionError::MissingInputWithEd25519Address)?,
```

Comment:
Not sure I like how it is currently. What about?
```rust
if { match &input_address {
    Address::Ed25519(_) | Address::ImplicitAccountCreation(_) => true,
    Address::Restricted(restricted) if restricted.address().is_ed25519() => true,
    _ => false
} } { ... } else { Err(InputSelectionError::MissingInputWithEd25519Address)? }
```
Maybe I would consider binding the match first to a `let is_valid_address`, but I'ld find that overall easier to read.

Reply:
Honestly think my version is more readable, same thing, less verbose 🤷🏻‍♂️

Reply:
okay you can keep it, but I think code should be explicit as possible and this is essentially somewhat hidden `if else`.

Reply:
I mean, literally all match are hidden if/else, that's syntactic sugar to make things shorter/prettier